src/ZoneServer/Skills/Handlers/Archer/Fulldraw.cs

```diff
@@ -0,0 +1,111 @@
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.Skills.SplashAreas;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.Characters;
+using Melia.Zone.World.Actors.CombatEntities.Components;
+using Yggdrasil.Extensions;
+using Yggdrasil.Logging;
+using static Melia.Zone.Skills.SkillUseFunctions;
+using Melia.Shared.Data.Database;
+
+namespace Melia.Zone.Skills.Handlers.Archer
+{
+	/// <summary>
+	/// Handler for the Archer skill Multishot.
+	/// </summary>
+	[SkillHandler(SkillId.Archer_Fulldraw)]
+	public class Fulldraw : ITargetSkillHandler, IDynamicCasted
+	{
+		/// <summary>
+		/// Called when the user starts casting the skill.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		public void StartDynamicCast(Skill skill, ICombatEntity caster)
+		{
+			Send.ZC_PLAY_SOUND(caster, "voice_war_atk_long_cast");
+		}
+
+		/// <summary>
+		/// Called when the user stops casting the skill.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		public void EndDynamicCast(Skill skill, ICombatEntity caster)
+		{
+			Send.ZC_STOP_SOUND(caster, "voice_war_atk_long_cast");
+		}
+
+		/// <summary>
+		/// Handles skill, damaging targets.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="originPos"></param>
+		/// <param name="farPos"></param>
+		public void Handle(Skill skill, ICombatEntity caster, ICombatEntity target)
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			var originPos = caster.Position;
+			var farPos = target.Position;
+
+			skill.IncreaseOverheat();
+			caster.TurnTowards(farPos);
+			caster.SetAttackState(true);
+
+			var splashParam = skill.GetSplashParameters(caster, originPos, farPos, length: 150, width: 10, angle: 0);
+			var splashArea = skill.GetSplashArea(SplashType.Square, splashParam);
+
+			Send.ZC_SKILL_READY(caster, skill, originPos, farPos);
+			Send.ZC_SKILL_MELEE_GROUND(caster, skill, farPos, null);
+			
+			this.Attack(skill, caster, splashArea);
+		}
+
+		/// <summary>
+		/// Executes the actual attack after a delay.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="splashArea"></param>
+		private async void Attack(Skill skill, ICombatEntity caster, ISplashArea splashArea)
+		{
+			var hitDelay = TimeSpan.FromMilliseconds(100);
+			var damageDelay = TimeSpan.FromMilliseconds(100);
+			var skillHitDelay = TimeSpan.Zero;
+
+			await Task.Delay(hitDelay);
+
+			var targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
+			var skillHits = new List<SkillHitInfo>();
+
+			foreach (var target in targets.LimitBySDR(caster, skill))
+			{
+				var skillHitResult = SCR_SkillHit(caster, target, skill);
+				target.TakeDamage(skillHitResult.Damage, caster);
```

Comment:
This is a bit of a tough one.  The main reason I didn't implement the knockback, despite seeing it in the old videos and in skill_bytool, is because the skill's description does not mention it.  Notably, when the video I saw that had the knockback was recorded, the skill did not yet seem to have the immobilize effect, so it's likely the effect was changed at some point prior to its removal.  Also, IMO, it doesn't really make a lot of sense for it to have both a knockback and an immobilize effect, and this also results in more overlap with Heavy Shot, which is a knockback attack.

Reply:
I see where you're coming from, but I mean... we have bytool, we have video evidence, and in the video I saw, the monsters did stop moving after the knock back. I do not think that we should go with custom implementations when we clearly know what the official behavior is supposed to be.

Your assumption that the skill changed over time also doesn't appear to be true. I just checked my data archives, and both the knock back and the hold effect have been present across all versions, starting with the first closed beta, meaning they've been present in every release version. Also, the description always said `[Skew] for X seconds`, and never said anything about knock back, suggesting to me that it always held targets, and that the knock back was simply omitted. That's far from unusual. Energy Bolt for example knocks enemies back as well, but doesn't say so in the description.

This seems like an open and shut case to me, don't you think?

Reply:
All right, I've added the knockback.  While we're on the subject, it's clear from old videos and Skill Bytool that the skill also inflicts a status called Link, which is actually kind of insane, it makes it so that any future damage taken by any of the targets of the skill is shared among all of the skill's targets, which I've also added, though currently its visual effect doesn't play properly (neither does the one for slow, probably ZC_Normal.PlayTextEffect has the wrong opcode).

Reply:
Thanks to Akinue's help, the visual for PlayTextEffect now does display.  It's possible this is not the right message / font, or maybe they changed it at some point.

Reply:
If you'd like, you can check my last couple commits to gain more insight into how `PlayTextEffect` works. Perhaps the right text style is among the ones I found.